tt_eager/tensor/tensor.cpp

```diff
@@ -208,7 +208,8 @@ void Tensor::deallocate(bool force) {
                 } else if constexpr (std::is_same_v<T, MultiDeviceHostStorage>) {
                     if (this->tensor_attributes.use_count() == 1) {
                         // Same logic as above for host tensors
-                        for (auto& current_buffer : storage.buffers) {
+                        for (int i = 0; i < storage.num_buffers(); i++) {
+                            auto& current_buffer = storage.get_buffer(i);
```

Comment:
We assume that `storage.buffers` does not change in size after we get the number of buffers. Is this safe?

Reply:
For `MultiDeviceHostStorage`, the _buffers_ vector is pre-populated (see worker based tensor constructor in `tensor.hpp`). The size should not change.